### PR TITLE
pip install libqfieldsync

### PR DESCRIPTION
**Pull request by m-kuhn**
By default installs libqfieldsync via pip
For local development, allows mounting a volume into the qgis container

Depends on https://github.com/opengisch/libqfieldsync/pull/49

### CODE REVIEW

File: docker-compose.override.local.yml [hunk 1]
```diff
@@ -125,6 +125,13 @@ services:
       exit 0;
       "
 
+  qgis:
+    build:
+      target: dev
+    volumes:
+# Uncomment to use a local libqfieldsync dev environment
+#      - /home/user/dev/libqfieldsync:/libqfieldsync
+
```
Comment (suricactus) [inline docker-compose.override.local.yml:134]: Can we use a envvar for this? I would prefer we never keep local modifications in the `docker-compose` files. So far all of these situations are handled with an envvar.

File: .gitignore [hunk 2]
```diff
@@ -1,14 +0,0 @@
-__pycache__/
```
Comment (suricactus) [inline .gitignore:1]: I guess it was a mistake that the global `.gitignore` was deleted. Can you please revert.

File: docker-compose.override.local.yml (hunk 1 above)
Comment (m-kuhn) [reply docker-compose.override.local.yml:134]: The first part before the `:` needs to point to a valid directory. We could by default mount some random directory from the source tree. Or a dummy volume. Both approaches should work but seem clumsy, that's why I didn't implement one of them in the first place. Do you see an option I forgot?

Comment (why-not-try-calmer) [reply docker-compose.override.local.yml]: added

**Comment by why-not-try-calmer** [comment]
I had some doubts about `pip install -e` but I must this is a good use-case!

**Comment by suricactus** [comment]
Closing, #851 fixed this.

**Pull request closed without merging**